phoenix-core/src/test/java/org/apache/phoenix/replication/reader/ReplicationLogProcessorTest.java

```diff
@@ -0,0 +1,720 @@
+        // Attempt to process non-existent file - should throw IOException
+        try {
+            replicationLogProcessor.processLogFile(localFs, nonExistentFilePath);
+            fail("Should throw IOException for non-existent file");
+        } catch (IOException e) {
+            // Expected behavior - non-existent file should cause IOException
+            assertTrue("Should throw IOException for non-existent file", true);
+        }
+    }
+
+    /**
+     * Tests batching logic when processing log files with mutations for multiple tables.
+     */
+    @Test
+    public void testProcessLogFileBatchingWithMultipleTables() throws Exception {
+        final Path multiTableBatchFilePath = new Path(testFolder.newFile("testMultiTableBatch").toURI());
+        final String table1Name = "T1_" + generateUniqueName();
+        final String table2Name = "T2_" + generateUniqueName();
+        final int batchSize = 4;
+        final int recordsPerTable = 3;
+        final int totalRecords = recordsPerTable * 2; // 6 total records
+        final int expectedBatchCalls = (totalRecords + batchSize - 1) / batchSize; // 2 calls
+        // Create log file with mutations for multiple tables
+        LogFileWriter writer = initLogFileWriter(multiTableBatchFilePath);
+        // Add mutations alternating between tables using LogFileTestUtil
+        for (int i = 0; i < recordsPerTable; i++) {
+            // Add mutation for table1
+            Mutation put1 = LogFileTestUtil.newPut("row1_" + i, (i * 2) + 1, (i * 2) + 1);
+            writer.append(table1Name, (i * 2) + 1, put1);
+            writer.sync();
+            // Add mutation for table2
+            Mutation put2 = LogFileTestUtil.newPut("row2_" + i, (i * 2) + 2, (i * 2) + 2);
+            writer.append(table2Name, (i * 2) + 2, put2);
+            writer.sync();
+        }
+        writer.close();
+        // Create processor with custom batch size and spy on it
+        Configuration testConf = new Configuration(conf);
+        testConf.setInt(ReplicationLogProcessor.REPLICATION_STANDBY_LOG_REPLAY_BATCH_SIZE, batchSize);
+
+        ReplicationLogProcessor replicationLogProcessor = new ReplicationLogProcessor(testConf, executorService);
+        // Validate that the batch size is correctly set
+        assertEquals("Batch size should be set correctly", batchSize, replicationLogProcessor.getBatchSize());
+        ReplicationLogProcessor spyProcessor = Mockito.spy(replicationLogProcessor);
+        // Mock the processReplicationLogBatch method
+        Mockito.doNothing().when(spyProcessor).processReplicationLogBatch(Mockito.any(Map.class));
+        // Process the log file
+        spyProcessor.processLogFile(localFs, multiTableBatchFilePath);
+        // Verify processReplicationLogBatch was called the expected number of times
+        Mockito.verify(spyProcessor, Mockito.times(expectedBatchCalls))
+            .processReplicationLogBatch(Mockito.any(Map.class));
```

Comment:
You can also use your spy to check if expected mutations were found in each invocation of processReplicationLogBatch, put1 in the first, put2 in the second, as a sanity check.